build_tools/travis/install.sh

```diff
@@ -17,23 +17,32 @@ echo 'List files from cached directories'
 echo 'pip:'
 ls $HOME/.cache/pip
 
-export CC=/usr/lib/ccache/gcc
-export CXX=/usr/lib/ccache/g++
-# Useful for debugging how ccache is used
-# export CCACHE_LOGFILE=/tmp/ccache.log
-# ~60M is used by .ccache when compiling from scratch at the time of writing
-ccache --max-size 100M --show-stats
+if [ $TRAVIS_OS_NAME = "linux" ]
+then
+	export CC=/usr/lib/ccache/gcc
+	export CXX=/usr/lib/ccache/g++
+	# Useful for debugging how ccache is used
+	# export CCACHE_LOGFILE=/tmp/ccache.log
+	# ~60M is used by .ccache when compiling from scratch at the time of writing
+	ccache --max-size 100M --show-stats
+fi
 
 make_conda() {
 	TO_INSTALL="$@"
     # Deactivate the travis-provided virtual environment and setup a
     # conda-based environment instead
-    deactivate
+    deactivate || :
```

Comment:
What does this mean?

Reply:
Apparently `:` is like pass. `deactivate` isn't applicable if we are in `language: generic`